### PR TITLE
Rora performance

### PR DESCRIPTION
**Pull request by toolslive**
- null io
- io patterns (fixed, random, stride)
- small improvement to osd_access to let only 1 client do the fetching
- remove some high frequency log statements

### CODE REVIEW

File: cpp/src/lib/osd_access.cc [hunk 1]
```diff
@@ -61,14 +61,38 @@ void OsdAccess::_remove_ctx(osd_t osd) {
...
 
-  _osd_infos.clear();
-  for (auto &p : infos) {
-    _osd_infos.emplace(p.first, std::move(p.second));
-  }
+void OsdAccess::update(Proxy_client& client) {
+    if(!_filling){
+        ALBA_LOG(INFO, "OsdAccess::update:: filling up");
+        {
+            std::lock_guard<std::mutex> f_lock(_filling_mutex);
```
Comment (domsj) [inline cpp/src/lib/osd_access.cc]: don't you want/need to hold the lock for the entire duration of the `if(!_filling){` block?

File: cpp/src/lib/osd_access.cc [hunk 2]
```diff
@@ -61,14 +61,38 @@ void OsdAccess::_remove_ctx(osd_t osd) {
...
-void OsdAccess::update(std::vector<std::pair<osd_t, info_caps>> &infos) {
-  ALBA_LOG(DEBUG, "OsdAccess::update");
-  std::lock_guard<std::mutex> lock(_osd_infos_mutex);
 
-  _osd_infos.clear();
-  for (auto &p : infos) {
-    _osd_infos.emplace(p.first, std::move(p.second));
-  }
+void OsdAccess::update(Proxy_client& client) {
+    if(!_filling){
```
Comment (domsj) [inline cpp/src/lib/osd_access.cc]: you may need to specify that you want to do a 'volatile' read here?
(that's the c# terminology, in c++ the stuff in std::memory_order might be related)
@redlicha can you review this part?

nit: please do a clang-format on this file.

Comment (domsj) [inline cpp/src/lib/osd_access.cc]: It's possible that 2 clients will enter this block simultaneously and still perform too much work.
This is typically solved by the following pattern (double checked locking):
```
if (need_to_do_sth) {
  take lock;
  if (need_to_do_sth) {
    do_thing();
    need_to_do_sth = false;
  } else {
    ...
  }
  release lock;
} else {
  ...
}

Comment (domsj) [inline cpp/src/lib/osd_access.cc]: just remembered that @redlicha is on pto. @cnanakos can you review please?

Comment (toolslive) [inline cpp/src/lib/osd_access.cc]: ok, double checked locking combined with std::atomic it will be.

**Comment by cnanakos** [comment]
You can directly use atomic CAS but this is fine also.

**Comment by toolslive** [comment]
master has plenty of formatting issues, which should be fixed by the latest format here.

**Pull request merged**